src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,129 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double DEFAULT_LOAD_FACTOR = 0.75;
+    private int arrayCapacity;
+    private int threshold;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        arrayCapacity = DEFAULT_CAPACITY;
+        table = new Node[arrayCapacity];
+        threshold = (int) (arrayCapacity * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size == threshold) {
+            resize();
+        }
+        add(key, value);
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        if (table[index] == null) {
+            return null;
+        }
+        if (equalsKey(table[index].key, key)) {
+            return table[index].value;
+        } else {
+            Node<K, V> currentNode = table[index];
+            while (currentNode != null) {
+                if (equalsKey(currentNode.key, key)) {
+                    return currentNode.value;
+                } else {
+                    currentNode = currentNode.next;
+                }
+            }
+        }
         return null;
```

Comment:
```suggestion
        Node<K, V> currentNode = table[index];
        while (currentNode != null) {
            if (Objects.equals(key, currentNode.key)) {
                return currentNode.value;
            }
            currentNode = currentNode.next;
        }
        return null;
```
посуті перший if ми обєднали з while на 35 строці
if на 28 строці і if на 31 роблять те саме що і while на 35 - тому можна це обєднати